server/src/main/java/org/elasticsearch/search/suggest/phrase/NoisyChannelSpellChecker.java

```diff
@@ -18,48 +18,34 @@
  */
 package org.elasticsearch.search.suggest.phrase;
 
-import org.apache.lucene.analysis.Analyzer;
 import org.apache.lucene.analysis.TokenStream;
 import org.apache.lucene.analysis.shingle.ShingleFilter;
 import org.apache.lucene.analysis.synonym.SynonymFilter;
 import org.apache.lucene.analysis.tokenattributes.TypeAttribute;
 import org.apache.lucene.codecs.TermStats;
-import org.apache.lucene.index.IndexReader;
 import org.apache.lucene.util.BytesRef;
 import org.apache.lucene.util.BytesRefBuilder;
-import org.apache.lucene.util.CharsRefBuilder;
 import org.elasticsearch.search.suggest.phrase.DirectCandidateGenerator.Candidate;
 import org.elasticsearch.search.suggest.phrase.DirectCandidateGenerator.CandidateSet;
 
-import java.io.CharArrayReader;
 import java.io.IOException;
 import java.util.ArrayList;
 import java.util.List;
 
-//TODO public for tests
-public final class NoisyChannelSpellChecker {
+final class NoisyChannelSpellChecker {
     public static final double REAL_WORD_LIKELIHOOD = 0.95d;
     public static final int DEFAULT_TOKEN_LIMIT = 10;
     private final double realWordLikelihood;
     private final boolean requireUnigram;
     private final int tokenLimit;
 
-    public NoisyChannelSpellChecker() {
-        this(REAL_WORD_LIKELIHOOD);
-    }
-
-    public NoisyChannelSpellChecker(double nonErrorLikelihood) {
-        this(nonErrorLikelihood, true, DEFAULT_TOKEN_LIMIT);
-    }
```

Comment:
I tend to lean towards having constructors with "default parameters" so that using the class doesn't require knowing that e.g. you should use `REAL_WORLD_LIKELIHOOD` and `DEFAULT_TOKEN_LIMIT`, but there's definitely benefits to having to pass them explicitly too. Not something I feel strongly enough to suggest, just thought it's worth mentioning

Reply:
I agree with this for user-facing or much used classe. Since this class is only used internally (I think) and even their very limited, I'd prefer removing the extra code though.